### PR TITLE
Transition `ToastMessages` to tailwind

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR transitions `ToastMessages` to tailwind. 

~~During this work, I stumbled upon a useful application of [Tailwind's `container` component class](https://tailwindcss.com/docs/container) that may be a sensible replacement for the styling-only `SidebarContent` component. I've commented its configuration in the tailwind configuration here, but will expand on it in some follow-up work (probably today).~~ <-- That work has been merged into this branch.

### User-visible changes

* `ToastMessage` now uses the shared `Card` component. The only directly-visible change from this is that it now has a stronger shadow on hover, like other “cards.”
* Padding around text has been adjusted to match other card-like components in the sidebar.
* At wide breakpoints, the edges of the toast messages are better aligned with “page content”
* For "notice" messages (I don't know that we even have any in the app at this time), the word "Notice: " is no longer pre-pended to message text; I didn't feel that was useful.

A success message looks like this:

image

An error message looks like this:

image

A (faked) notice message looks like this:

image

At wider screens, toast messages look like this:

image

Part of https://github.com/hypothesis/client/issues/4377

### CODE REVIEW

**Comment by lyzadanger** [comment]
Note: I've merged #4427 into this branch.

**Comment by robertknight** [comment]
In the context of the full page apps (notebook, single annotation), having the toast message fly the whole way across the screen until it reaches the the center feels a bit dramatic to me. If did want this kind of animation we'd probably want to tone it down for users with [prefers-used-motion](https://web.dev/prefers-reduced-motion/) active.

**Comment by lyzadanger** [comment]
> In the context of the full page apps (notebook, single annotation), having the toast message fly the whole way across the screen until it reaches the the center feels a bit dramatic to me. If did want this kind of animation we'd probably want to tone it down for users with [prefers-used-motion](https://web.dev/prefers-reduced-motion/) active. This animation already exists on master, although it is very slightly more subtle as it fades in whereas on this branch it seems fully opaque.

@robertknight At the risk of sounding too much like an acolyte, this is a place where Tailwind really shines :). I've pushed a commit that:

* Uses a fade instead of a fly-in animation when motion-reduction is preferred
* Uses fly-in animation when motion is OK and in narrow viewports
* Uses fade always at wider breakpoints

We can tune all of these effects later but I agree that the long-distance fly-in on wider screens is too much. 

Does this look OK to you?

**Review by robertknight — COMMENTED** [review]
On master the toast message fades out when clicked. On this branch nothing happens immediately when a toast message is clicked but then it disappears after a short delay. I assume this is an unintended result of a transition not working properly?

**Comment by lyzadanger** [comment]
> On master the toast message fades out when clicked. On this branch nothing happens immediately when a toast message is clicked but then it disappears after a short delay. I assume this is an unintended result of a transition not working properly?

Nice catch! Regression introduced by the last commit. Too many battling animation classes. This should be fixed now.

**Review by robertknight — APPROVED** [review]

File: src/sidebar/components/ToastMessages.js [hunk 1]
```diff
@@ -88,19 +106,37 @@ function ToastMessage({ message, onDismiss }) {
...
+            className={classnames(
+              'relative w-full container hover:cursor-pointer',
+              {
+                // Slide in from right in narrow viewports; fade in in
+                // larger viewports to toast message isn't flying too far
+                'motion-safe:animate-slide-in-from-right lg:animate-fade-in':
+                  !message.isDismissed,
+                // Only ever fade in if motion-reduction is preferred
+                'motion-reduce:animate-fade-in': !message.isDismissed,
+                'animate-fade-out': message.isDismissed,
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/ToastMessages.js:130]: Fade in and out are still applied in `motion-reduce`.

**Pull request merged**